x/mongo/driver/topology/connection.go

```diff
@@ -712,13 +713,30 @@ func (c *Connection) ID() string {
 	return c.id
 }
 
+// ServerConnectionID returns the server connection ID of this connection.
+func (c *Connection) ServerConnectionID() *int32 {
+	c.mu.RLock()
+	defer c.mu.RUnlock()
+	if c.connection == nil {
+		return nil
+	}
+	return c.serverConnectionID
```

Comment:
Are there any other cases where we try to access the underlying `*connection` and need nil checks? If we want to make all calls to the underlying `*connection` synchronized and nil-safe, we should un-embed the `*connection` field so that you can't call through to functions on the underlying `*connection`.

E.g.
```go
type Connection struct {
	connection *connection
	// ...
```